### PR TITLE
run clippy

### PR DESCRIPTION
**Pull request by Meow**
i ran `cargo clippy` and it was unhappy so i made it happy

### CODE REVIEW

**Comment by liamwhite** [comment]
Do you think we can send this upstream to https://github.com/kivikakk/comrak so that we don't have to keep reapplying it on every rebase?

**Comment by Meow** [comment]
> Do you think we can send this upstream to https://github.com/kivikakk/comrak so that we don't have to keep reapplying it on every rebase?

They are on edition 2018 and idk if what clippy pointed out is only applicable to 2021 or not.

File: src/nodes.rs
Comment (liamwhite) [inline src/nodes.rs:1]: All changes upstream in https://github.com/kivikakk/comrak/commit/3ecb988e450caf86f2529df08d8a969aa4224d96

File: src/parser/mod.rs
Comment (liamwhite) [inline src/parser/mod.rs:1]: All changes upstream in
https://github.com/kivikakk/comrak/commit/3ecb988e450caf86f2529df08d8a969aa4224d96
https://github.com/kivikakk/comrak/commit/f20cd772b26b74cd824de7367c02e65acc56bbf5

File: src/parser/inlines.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ impl<'a, 'r, 'o, 'd, 'i, 'c, 'subj> Subject<'a, 'r, 'o, 'd, 'i, 'c, 'subj> {
                 let start_column = self.pos;
                 self.pos += 1;
 
-                let id = self.scan_image_mention_id().unwrap_or_else(|| vec![]);
+                let id = self.scan_image_mention_id().unwrap_or_default();
```
Comment (liamwhite) [inline src/parser/inlines.rs:230]: This one is valid because it's from our own changes

**Pull request merged**